scenes/maze/room_trigger.gd

```diff
@@ -27,6 +36,11 @@ func _process(delta: float) -> void:
 	
 func _on_body_entered(body: Node2D) -> void: 
 	player_entered_room = true
+	if !is_room_solved:
+		_show_hint()
 		
 func _on_body_exited(body: Node2D) -> void:
 	player_entered_room = false
+	
+func _show_hint() -> void:
+	DialogManager.start_dialog(player, hints)
```

Comment:
in the final version, i think we should add a timer that only shows the hint after a few seconds or so

Reply:
I totally agree. I'm not sure on how to actually implement it though (e.g., if we wait x seconds before showing the hint and the player already left the room, we need to prevent the hints to show up at the wrong place)